core/src/main/java/com/scalar/db/api/TransactionCrudOperable.java

```diff
@@ -44,63 +46,129 @@ public interface TransactionCrudOperable {
-   * @throws UnsatisfiedConditionException if the mutation condition is not satisfied
+   * @throws UnsatisfiedConditionException if a condition is specified, and if the condition is not
+   *     satisfied or the entry does not exist
    */
   void delete(Delete delete)
       throws CrudConflictException, CrudException, UnsatisfiedConditionException;
 
   /**
    * Deletes entries from the storage through a transaction with the specified list of {@link
-   * Delete} commands.
+   * Delete} commands. If a condition is specified in the {@link Delete} command, and if the
+   * condition is not satisfied or the entry does not exist, it throws {@link
+   * UnsatisfiedConditionException}.
    *
    * @param deletes a list of {@code Delete} commands
    * @throws CrudConflictException if the transaction CRUD operation fails due to transient faults
    *     (e.g., a conflict error). You can retry the transaction from the beginning
    * @throws CrudException if the transaction CRUD operation fails due to transient or nontransient
    *     faults. You can try retrying the transaction from the beginning, but the transaction may
    *     still fail if the cause is nontranient
-   * @throws UnsatisfiedConditionException if the mutation condition is not satisfied
+   * @throws UnsatisfiedConditionException if a condition is specified, and if the condition is not
+   *     satisfied or the entry does not exist
+   * @deprecated As of release 3.13.0. Will be removed in release 5.0.0. Use {@link #mutate(List)}
+   *     instead.
    */
+  @Deprecated
   void delete(List<Delete> deletes)
       throws CrudConflictException, CrudException, UnsatisfiedConditionException;
 
+  /**
+   * Inserts an entry into the storage through a transaction with the specified {@link Insert}
+   * command. If the entry already exists, a conflict error occurs. Note that the location where the
+   * conflict error is thrown depends on the implementation of the transaction manager. This method
+   * may throw {@link CrudConflictException}. Alternatively, {@link DistributedTransaction#commit()}
+   * or {@link TwoPhaseCommitTransaction#prepare()} may throw {@link CommitConflictException} or
+   * {@link PreparationConflictException} respectively in case of a conflict error.
+   *
+   * @param insert a {@code Insert} command
+   * @throws CrudConflictException if the transaction CRUD operation fails due to transient faults
+   *     (e.g., a conflict error). You can retry the transaction from the beginning
+   * @throws CrudException if the transaction CRUD operation fails due to transient or nontransient
+   *     faults. You can try retrying the transaction from the beginning, but the transaction may
+   *     still fail if the cause is nontranient
+   */
+  void insert(Insert insert) throws CrudConflictException, CrudException;
+
+  /**
+   * Inserts or updates an entry in the storage through a transaction with the specified {@link
```

Comment:
In this case, since we need to use the same phrase consistently, the only way I can think of to get around this is to structure the sentence as follows:

```suggestion
   * Inserts an entry into or updates an entry in the underlying storage through a transaction with the specified {@link
```